conductor/src/client.cpp

```diff
@@ -0,0 +1,111 @@
+
+#include <cerrno>
+
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "client.hpp"
+#include "action.hpp"
+#include "actions.hpp"
+#include "isolate.hpp"
+
+
+
+namespace cocaine { namespace isolate { namespace conductor {
+
+container_t::container_t(shared_ptr<client_t> parent,
+                         std::string container_id,
+                         std::string name,
+                         dynamic_t profile,
+                         std::string stdout_path):
+    m_terminated(false),
+    m_parent(parent),
+    m_container_id(container_id),
+    m_name(name),
+    m_profile(profile),
+    m_stdout_path(stdout_path),
+    m_fd(-1)
+{}
+    
+void
+container_t::terminate(){
+    if (!m_terminated){
+        m_terminated = true;
+        // shared_ptr<action::action_t> action(
+        //     new action::terminate_t (m_parent, m_container_id)
+        // );
+        auto action = make_shared<action::terminate_t>(
+            m_parent,
+            m_container_id,
+            m_name,
+            m_profile
+        );
+
+        m_parent->enqueue(action);
+    }
+}
+
+void
+container_t::attach (){
+    COCAINE_LOG_DEBUG(m_parent->m_log, "attaching to container stdout fifo at %s", m_stdout_path);
+    auto fd = ::open(m_stdout_path.c_str(), O_RDONLY) == -1;
```

Comment:
Abstractions for fags, only low-level, only hardcore!

I see that this fd lives forever, blowing up our resources.